main.cc

```diff
@@ -1027,12 +1027,18 @@ int main(int ac, char** av) {
                 lb->stop_broadcasting().get();
             });
             supervisor::notify("starting cf cache hit rate calculator");
-            cf_cache_hitrate_calculator.start(std::ref(db), std::ref(cf_cache_hitrate_calculator)).get();
-            auto stop_cache_hitrate_calculator = defer_with_log_on_error([&cf_cache_hitrate_calculator] {
+
+            auto stop_cache_hitrate_calculator = defer_with_log_on_error([&cf_cache_hitrate_calculator, &db] {
```

Comment:
This should be constructed after `cf_cache_hitrate_calculator.start()`, becasue if start() fails, we must not attempt to stop.

Reply:
Ok. Sending fix now.